ert_shared/models/base_run_model.py

```diff
@@ -28,6 +30,29 @@ class ErtRunError(Exception):
     pass
 
 
+class _LogAggregration(logging.Handler):
+    """Logging handler which aggregates the log messages"""
+
+    def __init__(self) -> None:
+        self.messages: List[str] = []
+        super().__init__()
+
+    def emit(self, record: logging.LogRecord) -> None:
+        self.messages.append(record.getMessage())
+
+
+@contextmanager
+def captured_logs(level: int = logging.ERROR) -> Iterator[_LogAggregration]:
+    handler = _LogAggregration()
+    root_logger = logging.getLogger()
+    handler.setLevel(level)
+    root_logger.addHandler(handler)
+    try:
+        yield handler
+    finally:
+        root_logger.removeHandler(handler)
```

Comment:
It's an interesting way of setting up some sort of localised capturing of logs; contained and self-cleaning. Can't tell if this is a proposed way of doing it, but first impression is that I like it!